meta-mbl-distro/scripts/create-update-payload

```diff
@@ -0,0 +1,191 @@
+
+
+"""
+This script will create a bootloader update payload tar archive.
+
+You can specify a compression type for the tar archive. The default is gz and
+the available options are gz, bz2 and xz.
+
+To use the script you can either pass in a space separated list of file names
+or a list of bootloader slot numbers. We then map the file names of the update
+components to the corresponding bitbake variables, which are named according
+to the bootloader slot the file will end up in.
+
+We rename the files MBL_WKS_BOOTLOADER[n] so the update scripts on the target
+only have to know which bootloader slot they're updating, and are generic for
+any MACHINE target.
+
+The script assumes the files you want to archive are in DEPLOY_DIR_IMAGE in
+your MBL workspace.
+
+The script supports multi-component or single component payloads.
+
+Example invocation:
+
+create-update-payload --bootloader-slots 1 2 --output-tar-path tmp/payload.tar
+
+This will create the update payload containing the bootloader 1 and 2
+components at tmp/payload.tar
+"""
+
+import pathlib
+import re
+import shutil
+import sys
+import tarfile
+import tempfile
+
+from itertools import product
+
+# Set the PYTHONPATH so we can import openembedded scripts/lib modules.
+# This script can only be executed from the
+# meta-mbl-distro/scripts folder in the MBL workspace.
+oe_scripts_path = (
+    pathlib.Path(__file__).parent.parent.parent.parent.resolve()
+    / "openembedded-core"
+    / "scripts"
+)
+
+lib_path = oe_scripts_path / "lib"
+sys.path.insert(0, str(lib_path))
+
+import scriptpath
+import scriptutils
+
+# This is needed so we can import bitbake modules.
+bitbakepath = scriptpath.add_bitbake_lib_path()
+
+import bb.tinfoil
+import argparse_oe
+
+
+def parse_args():
+    """Parse the command line arguments."""
+    parser = argparse_oe.ArgumentParser()
+    parser.add_argument("-f", "--filenames", nargs="+")
+    parser.add_argument("-b", "--bootloader-slots", nargs="+")
+    parser.add_argument(
+        "-o",
+        "--tar-output-path",
+        type=lambda x: pathlib.Path(x).absolute(),
+        default="payload.tar",
+    )
+    parser.add_argument(
+        "-c", "--compression-format", default="gz", choices=["gz", "bz2", "xz"]
+    )
+    args = parser.parse_args()
+
+    if args.filenames is None and args.bootloader_slots is None:
+        raise Exception()
+
+    return args
+
+
+def match_varnames_to(filenames, tinfoil=None):
+    """
+    Match bitbake variable names to the given file names.
+
+    Parse the bitbake config data using the tinfoil API and find the
+    variable which is bound to the file name.
+
+    Return a dict with the variable names as keys and
+    file names as values.
+
+    :param filenames list: List of file names to look for.
+    :param tinfoil Tinfoil: bitbake tinfoil object.
+    """
+    return {
+        cvar.split("_FILENAME")[0]: fname
+        for cvar, fname in product(tinfoil.config_data, filenames)
```

Comment:
There are a lot of `cvar`s here, right? Should we filter for starting with "MBL_WKS" before we create the product?